packages/core/src/components/Dropdown/Dropdown.jsx

```diff
@@ -0,0 +1,169 @@
+import FormLabel from '../FormLabel/FormLabel';
+import PropTypes from 'prop-types';
+import React from 'react';
+import classNames from 'classnames';
+import uniqueId from 'lodash.uniqueid';
+
+/**
+ * A `Dropdown` component can be used to render an HTML `select` menu.
+ * Any _undocumented_ props that you pass to this component will be passed
+ * to the `select` element, so you can use this to set additional attributes if
+ * necessary.
+ *
+ * Class-based component gives flexibility for active focus management
+ * by allowing refs to be passed.
+ */
+export class Dropdown extends React.PureComponent {
+  componentDidMount() {
+    if (this.props.focusTrigger) {
+      this.loader && this.loader.focus();
+    }
+  }
+
+  id() {
+    if (!this._id) {
+      // Cache the ID so we're not regenerating it on each method call
+      this._id = uniqueId(`select_${this.props.name}_`);
+    }
+    return this._id;
+  }
+
+  render() {
+    /* eslint-disable prefer-const */
+    let {
+      // Using let rather than const since we sometimes rewrite id
+      className,
+      errorMessage,
+      fieldClassName,
+      fieldRef,
+      focusTrigger,
+      hint,
+      inversed,
+      label,
+      labelClassName,
+      options,
+      requirementLabel,
+      size,
+      ...selectProps
+    } = this.props;
+    /* eslint-enable prefer-const */
+
+    const classes = classNames(className);
+    const fieldClasses = classNames(
+      'ds-c-field',
+      { 'ds-c-field--inverse': inversed },
+      size && `ds-c-field--${size}`,
+      fieldClassName
+    );
+
+    const optionElements = options.map(option => (
+      <option key={option.value}>{option.label}</option>
+    ));
+
+    return (
+      <div className={classes}>
+        <FormLabel
+          className={labelClassName}
+          component={'label'}
+          errorMessage={errorMessage}
+          fieldId={this.id()}
+          hint={hint}
+          requirementLabel={requirementLabel}
+          inversed={inversed}
+        >
+          {label}
+        </FormLabel>
+        <select
+          className={fieldClasses}
+          id={this.id()}
+          /* eslint-disable no-return-assign */
+          ref={focusTrigger ? loader => (this.loader = loader) : fieldRef}
```

Comment:
Could it be a problem to make focus management and letting the parent specify a ref function be mutually exclusive? I think it might cause unexpected behavior for the user who expects his or her `fieldRef` function to be executed. Something like
```js
ref => {
  if (focusTrigger) {
    this.selectRef = ref;
  }
  this.props.fieldRef(ref);
}
```

Reply:
I agree. This looks like a better approach on the whole. I wrote the ternary logic nearly a year ago to fix a specific problem. Now that the design system is gaining wider adoption, defensive patterns like this are more sustainable.